pkg/meta/model/job_args.go

```diff
@@ -425,3 +425,66 @@ func UpdateRenameTableArgs(job *Job) error {
 	}
 	return nil
 }
+
+// CheckConstraintArgs is the arguments for both AlterCheckConstraint and DropCheckConstraint job.
+type CheckConstraintArgs struct {
+	ConstraintName pmodel.CIStr `json:"constraint_name,omitempty"`
+	Enforced       bool         `json:"enforced,omitempty"`
+}
+
+func (a *CheckConstraintArgs) fillJob(job *Job) {
+	if job.Version == JobVersion1 {
+		job.Args = []any{a.ConstraintName, a.Enforced}
+		return
+	}
+	job.Args = []any{a}
+}
+
+// GetCheckConstraintArgs gets the AlterCheckConstraint args.
+func GetCheckConstraintArgs(job *Job) (*CheckConstraintArgs, error) {
+	if job.Version == JobVersion1 {
+		var (
+			constraintName pmodel.CIStr
+			enforced       bool
+		)
+		err := job.DecodeArgs(&constraintName, &enforced)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		return &CheckConstraintArgs{
+			ConstraintName: constraintName,
+			Enforced:       enforced,
+		}, nil
+	}
+
+	return getOrDecodeArgsV2[*CheckConstraintArgs](job)
+}
+
+// AddCheckConstraintArgs is just a wrapper for ConstraintInfo.
+// It is used to maintain the consistency of all the arguments
```

Comment:
what's this mean?

Reply:
Since `add check constraints` only need one argument `ConstraintInfo`, we  can use it directly in V2. But I don't want to add function for `ConstraintInfo`.

Reply:
Maybe I can use type alias instead?

Reply:
even it's the only argument, we still need wrap it to make it better for later extension

and we need keep all XXargs in a uniform way, we need implement JobArgs interface here, but we shouldn't do it for ConstraintInfo

Reply:
please remove this comment

Reply:
or you can merge them with CheckConstraintArgs, like this https://github.com/pingcap/tidb/pull/56163/files#diff-0a328bfc50df4984d8b701f6cfb588c8bc675992e32e00ae05f2d900147ea8dfR375-R386

Reply:
if we save constraintName inside ConstraintInfo, they are nearly the same